content/en/docs/HOWTOs/EventFeeds/TranslationHowTo.md

```diff
@@ -0,0 +1,638 @@
+---
+title: "Writing an XSLT Translation"
+linkTitle: "Writing a Translation"
+#weight:
+date: 2022-06-07
+tags:
+  - processing
+  - feed
+description: >
+  This HOWTO will take you through the production of an XSLT for a feed, including issues such as event filtering, common errors and testing.
+---
+
+<!-- Created with Stroom v6.1-beta.16 -->
+
+{{% see-also %}}
+[HOWTO - Event Processing]({{< relref "ProcessingHowTo" >}})
+{{% /see-also %}}
+
+
+## Introduction
+
+This document is intended to explain how and why to produce a translation within stroom and how the translation fits into the overall processing within stroom.
+It's not intended as an XSLT tutorial so a basic XSLT knowledge must be assumed.
+The document will contain potentially useful XSLT fragments to show how certain processing activities can be carried out.
+As with most programming languages, there are likely to be multiple ways of producing the same end result with different degrees of complexity and efficiency.
+Examples here may not be the best for all situations but do reflect experience built up from many previous translation jobs.
+
+The document should be read in conjunction with other online stroom documentation, in particular [Event Processing]({{< relref "ProcessingHowTo" >}}).
+
+
+## Translation Overview
+
+The translation process for raw logs is a multi-stage process defined by the processing pipeline:
+
+### Parser
+
+The parser takes raw data and converts it into an intermediate XML document format.
+This is only required if source data is not already within an XML document.
+There are various standard parsers available (although not all may be available on a default stroom build) to cover the majority of standard source formats such as CSV, TSV, CSV with header row and XML fragments.
+
+The language used within the parser is defined within an XML schema located at `XML Schemas / data-splitter / data-splitter v3.0` within the tree browser.
+The language can be quite complex so if non-standard format logs are being parsed, it may be worth speaking to your stroom sysadmin team to at least get an initial parser configured for your data.
+
+Stroom also has a built-in parser for JSON fragments.
+This can be set either by using the _combinedParser_ and setting the `type` property to JSON or preferably by just using the _jsonParser_.
```

Comment:
You can use the pipe element shortcodes for these, see https://gchq.github.io/stroom-docs/7.0/community/documentation/style-guide/using-images/#pipeline-elements